### PR TITLE
move logic that is not specific to gromacs into hooks functions

### PR DESCRIPTION
**Pull request by smoors**
fixes https://github.com/EESSI/test-suite/issues/18

### CODE REVIEW

**Comment by boegel** [comment]
@smoors README still mentions `singlenode`, should be changed too?

File: eessi/reframe/eessi_utils/hooks.py [hunk 1]
```diff
@@ -93,3 +93,46 @@ def assign_one_task_per_gpu(test: rfm.RegressionTest) -> rfm.RegressionTest:
...
+    filter valid_systems by device type,
+        unless specified with --setvar valid_systems=<comma-separated-list>
+    """
+    if not test.valid_systems:
+        is_cuda_module = utils.is_cuda_required_module(test.module_name)
+        valid_systems = ''
+
+        if is_cuda_module and device_type == 'gpu':
+            # CUDA modules and when using a GPU for non-bonded interactions require partitions with 'gpu' feature
+            valid_systems = '+gpu'
```
Comment (casparvl) [inline eessi/reframe/eessi_utils/hooks.py:109]: Should use `'+%s' % eessi_utils.features.GPU_GENERIC`

File: eessi/reframe/eessi_utils/hooks.py [hunk 2]
```diff
@@ -93,3 +93,46 @@ def assign_one_task_per_gpu(test: rfm.RegressionTest) -> rfm.RegressionTest:
...
+
+        elif device_type == 'cpu':
+            # Non-bonded interactions on the CPU require partitions with 'cpu' feature
```
Comment (casparvl) [inline eessi/reframe/eessi_utils/hooks.py:112]: Test-specific comment

**Review by casparvl — CHANGES_REQUESTED** [review]
I've put a lot of comments. I suggest that in this PR, we only change stuff that is immediately relevant for the GROMACS test (function names, arguments), or trivial fixes (double indentation of docstring).

The replacement of hardcoded strings by constants from `eessi_utils.features` should probably be something we do in a separate PR.

File: eessi/reframe/eessi_utils/hooks.py (hunk 1 above)
Comment (smoors) [reply eessi/reframe/eessi_utils/hooks.py:109]: that's also for a separate PR

**Comment by smoors** [comment]
issue created for replacing strings with constants https://github.com/EESSI/test-suite/issues/30

**Review by boegel — APPROVED** [review]
lgtm

**Review by casparvl — APPROVED** [review]
Lgtm now

**Pull request merged**